### ISSUE TITLE
Problem: Bag with diacritics in bag-info.txt fails at Job: Verify bag, and restructure for compliance

### ISSUE THREAD

**Issue by evelynPM** [archivematica/Issues#977]
**Expected behaviour**
A bag with diacritics in bag-info.txt can be ingested and processed into an AIP.

**Current behaviour**
The bag fails at Job: Verify bag, and restructure for compliance:

```
`'utf8' codec can't decode byte 0xf4 in position 0: invalid continuation byteTraceback (most recent call last):
  File "/opt/archivematica/archivematica/src/MCPClient/lib/job.py", line 111, in JobContext
    yield
  File "/opt/archivematica/archivematica/src/MCPClient/lib/clientScripts/verify_and_restructure_transfer_bag.py", line 147, in call
    if not bag.is_valid(target, printfn=job.pyprint):
  File "/usr/lib/archivematica/archivematicaCommon/bag.py", line 26, in is_valid
    bag = Bag(path)
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/local/lib/python2.7/site-packages/bagit.py", line 305, in __init__
    self._open()
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/local/lib/python2.7/site-packages/bagit.py", line 371, in _open
    self.info = _load_tag_file(info_file_path, encoding=self.encoding)
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/local/lib/python2.7/site-packages/bagit.py", line 1166, in _load_tag_file
    for name, value in _parse_tags(tag_file):
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/local/lib/python2.7/site-packages/bagit.py", line 1193, in _parse_tags
    for num, line in enumerate(tag_file):
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/lib/python2.7/codecs.py", line 701, in next
    return self.reader.next()
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/lib/python2.7/codecs.py", line 632, in next
    line = self.readline()
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/lib/python2.7/codecs.py", line 547, in readline
    data = self.read(readsize, firstline=True)
  File "/usr/share/archivematica/virtualenvs/archivematica-mcp-client/lib/python2.7/codecs.py", line 494, in read
    newchars, decodedbytes = self.decode(data, self.errors)
UnicodeDecodeError: 'utf8' codec can't decode byte 0xf4 in position 0: invalid continuation byte`
```

**Steps to reproduce**
Ingest a bag with diacritics in the bag-info.txt, such as Côté.

**Your environment (version of Archivematica, operating system, other relevant details)**
AM 1.10.1. Note: this appears to be a regression from 1.9.x

---

**For Artefactual use:**

Before you close this issue, you must check off the following:

- [ ] All pull requests related to this issue are properly linked
- [ ] All pull requests related to this issue have been merged
- [ ] A testing plan for this issue has been implemented and passed (testing plan information should be included in the issue body or comments)
- [ ] Documentation regarding this issue has been written and merged
- [ ] Details about this issue have been added to the release notes (if applicable)

**Comment by replaceafill** [comment]
Sharing some on this.

I set up a virtual environment with `bagit-python 1.7.0` and found that it has decoding troubles (same error that the description) validating the bag when there's a mismatch like this:

1. The `bagit.txt` file specifies an UTF-8 encoding like this:

```
Tag-File-Character-Encoding: UTF-8
```

2. And the `bag-info.txt` file has an encoding like this:

```
$ file bag-info.txt 
bag-info.txt: ISO-8859 text, with CRLF line terminators
```

Changing either the encoding of the `bag-info.txt` file with `iconv` like this:

```
iconv -f ISO-8859-1 bag-info.txt -t UTF-8 -o bag-info.txt
```

or setting a different encoding in `bagit.txt` like this:

```
Tag-File-Character-Encoding: LATIN-1
```

made Evelyn's bag to validate with `bagit-python` (both of these changes required updating the md5 checksums in `tagmanifest-md5.txt`).

I also installed our [bagit package](https://packages.archivematica.org/1.10.x/ubuntu-externals/pool/main/b/bagit/bagit_4.12.3-1~18.04_amd64.deb) and found that `bagit-java` doesn't have this problem:

```
$ bag verifyvalid .
Result is true.
```
